components/centraldashboard/public/components/dashboard-view.js

```diff
@@ -119,20 +119,41 @@ export class DashboardView extends PolymerElement {
      * Observer for platformInfo property
      */
     _platformInfoChanged() {
-        if (this.platformInfo && this.platformInfo.providerName === 'gce') {
-            this.platformName = 'GCP';
-            const pieces = this.platformInfo.provider.split('/');
-            let gcpProject = '';
-            if (pieces.length >= 3) {
-                gcpProject = pieces[2];
-            }
-            this.platformLinks = GCP_LINKS.map((l) => {
-                return {
-                    text: l.text,
-                    link: l.link + gcpProject,
-                };
-            });
+        if (!this.platformInfo || this.platformInfo.providerName != 'gce') {
+            // return;
         }
+        this.platformName = 'GCP';
+        const pieces = this.platformInfo.provider.split('/');
+        let gcpProject = '';
+        if (pieces.length >= 3) {
+            gcpProject = pieces[2];
+        }
+        this.platformLinks = GCP_LINKS.map((l) => {
+            return {
+                text: l.text,
+                link: l.link + gcpProject,
+            };
+        });
+    }
+    /**
+     * Takes a platform abbrev and converts it to a fully qualified name
+     * @param {string} platform
+     * @return {string}
+     */
+    headerForPlatform(platform) {
+        return {
```

Comment:
I think both this and the `logoForPlatform` method could be eliminated in favor of a statically defined configuration object for each platform. They could be stored in a map keyed by the platform name and then assigned as a component property when the response comes back from the server.

This would let us eliminate all one but one of the platform properties and then just refer to the `platform` object's properties in the template. Then, adding a new platform is just a matter of defining the logo, links, header, etc. What do you think?

Reply:
Fair enough, I'll go ahead and add it in there! That seems to be a better approach, so all information is centralized. And it scales better as we add more fields.

Unrelated: do you know why the platform info api locally takes 10+ seconds to respond?

Reply:
Figured out it's because of network timeouts since I cannot connect to the cluster.